### PR TITLE
add support for kueue v0.9.0 to enable TAS

### PR DESCRIPTION
**Pull request by ighosh98**
Added support for [kueue v0.9.0](https://github.com/kubernetes-sigs/kueue/tree/release-0.9) which was released recently with TAS enabled for all clusters. As aligned offline, Kueue-v0.9.0 would now be the default Kueue version choice for customers. 

[kueue-v0.9.0](https://gist.githubusercontent.com/mimowo/f27a00a0fa0e6838016460a9f95b613c/raw/e6e8bc30effa846e0443064b9cd7f49008530d1b/kueue_0.9_tas_enabled.yaml) has TAS enabled in alpha phase. Have added necessary tolerations to kueue-v0.9.0 file to make the manifest compatible with cluster-toolkit. 

### Testing Information
Testing was done on a2-high clusters. 

Ran the command:
`kubectl get pods -o custom-columns="Name:.metadata.name,Host:.spec.nodeSelector.cloud\.google\.com/gce-topology-host" | sort -k2`
Job Annotation:
```
        kueue.x-k8s.io/podset-required-topology: "cloud.google.com/gce-topology-host"
```
Output:
Was testing host topology compliance required job. Observe that all pods are on the same host:
```
tas-sample-small-required-host-gqm74-0-sksh6   2131055a3f84f1c773b6cddca39d1a24
tas-sample-small-required-host-gqm74-1-gls2s   2131055a3f84f1c773b6cddca39d1a24
tas-sample-small-required-host-gqm74-2-7vchk   2131055a3f84f1c773b6cddca39d1a24
tas-sample-small-required-host-gqm74-3-rnhqv   2131055a3f84f1c773b6cddca39d1a24
tas-sample-small-required-host-x72sr-0-5bh4v   2131055a3f84f1c773b6cddca39d1a24
tas-sample-small-required-host-x72sr-1-h7b9k   2131055a3f84f1c773b6cddca39d1a24
tas-sample-small-required-host-x72sr-2-c4prq   2131055a3f84f1c773b6cddca39d1a24
tas-sample-small-required-host-x72sr-3-gwkb9   2131055a3f84f1c773b6cddca39d1a24
Name                                           Host 
```

### Submission Checklist

NOTE: Community submissions can take up to 2 weeks to be reviewed.

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cluster Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: modules/management/kubectl-apply/variables.tf [hunk 1]
```diff
@@ -41,15 +41,15 @@ variable "kueue" {
   description = "Install and configure [Kueue](https://kueue.sigs.k8s.io/docs/overview/) workload scheduler. A configuration yaml/template file can be provided with config_path to be applied right after kueue installation. If a template file provided, its variables can be set to config_template_vars."
   type = object({
     install              = optional(bool, false)
-    version              = optional(string, "v0.8.1")
+    version              = optional(string, "v0.9.0")
```
Comment (ankitkinra) [inline modules/management/kubectl-apply/variables.tf:44]: Let's not update the default , but use the version we want from the blueprint

File: modules/management/kubectl-apply/variables.tf [hunk 2]
```diff
@@ -41,15 +41,15 @@ variable "kueue" {
...
     config_path          = optional(string, null)
     config_template_vars = optional(map(any), null)
   })
   default = {}
 
   validation {
-    condition     = !var.kueue.install || contains(["v0.8.1"], var.kueue.version)
-    error_message = "Supported version of Kueue is v0.8.1"
+    condition     = !var.kueue.install || contains(["v0.9.0", "v0.8.1"], var.kueue.version)
+    error_message = "Default Supported version of Kueue is v0.9.0. Cluster toolkit also supports v0.8.1"
```
Comment (ankitkinra) [inline modules/management/kubectl-apply/variables.tf:52]: nit: Update error message to just output supported versions.

File: modules/management/kubectl-apply/variables.tf (hunk 1 above)
Comment (ighosh98) [reply modules/management/kubectl-apply/variables.tf:44]: Could you please explain why we should not update the default? My rationale for updating was that the changes should be backward compatible across versions and TAS support is independent of the node we choose.

File: modules/management/kubectl-apply/variables.tf (hunk 2 above)
Comment (ighosh98) [reply modules/management/kubectl-apply/variables.tf:52]: Updating this

File: modules/management/kubectl-apply/variables.tf (hunk 1 above)
Comment (annuay-google) [reply modules/management/kubectl-apply/variables.tf:44]: I agree with you on all versions being backward compatible and updating the default eventually @ighosh98. We should wait till we know for a fact that 0.9.0 is stable though

File: modules/management/kubectl-apply/variables.tf [hunk 3]
```diff
@@ -41,15 +41,15 @@ variable "kueue" {
...
+    version              = optional(string, "v0.9.0")
     config_path          = optional(string, null)
     config_template_vars = optional(map(any), null)
   })
   default = {}
 
   validation {
-    condition     = !var.kueue.install || contains(["v0.8.1"], var.kueue.version)
-    error_message = "Supported version of Kueue is v0.8.1"
+    condition     = !var.kueue.install || contains(["v0.9.0", "v0.8.1"], var.kueue.version)
```
Comment (annuay-google) [inline modules/management/kubectl-apply/variables.tf:51]: Store this in a list, and use it in the error message as well as the condition

Comment (ighosh98) [reply modules/management/kubectl-apply/variables.tf:51]: Requesting clarification: Isn't ["v0.9.0", "v0.8.1"] already a list? Are you recommending that we should store it in a new variable and then use it elsewhere?

Comment (annuay-google) [reply modules/management/kubectl-apply/variables.tf:51]: Yes, use a new variable and avoid the hardcoded versions in the error message. Use the variable in the error message

**Review by annuay-google — APPROVED** [review]
LGTM

**Pull request merged**